### PR TITLE
fix(xref/search): find concepts by varying term in multiple passes

### PR DESCRIPTION
**Pull request by sidvishnoi**
Fixes https://github.com/w3c/respec-web-services/issues/216
Closes https://github.com/w3c/respec-web-services/pull/217 (supersedes)

### CODE REVIEW

File: routes/xref/lib/search.ts [hunk 1]
```diff
@@ -118,34 +121,44 @@ function filter(query: Query, store: Store, options: Options) {
...
-    for (const altTerm of textVariations(term)) {
-      if (altTerm in store.byTerm) {
-        termData = store.byTerm[altTerm];
-        break;
-      }
-    }
+
+  if (shouldTreatAsConcept) {
+    const term = inputTerm.toLowerCase();
+    return (function* () {
```
Comment (marcoscaceres) [inline routes/xref/lib/search.ts:149]: Nit: can we put the generators as curried functions maybe? Just a bit easier on the eyes.

**Review by marcoscaceres — APPROVED** [review]
cool stuff! Just a style nit with the them generators functions.

File: routes/xref/lib/search.ts (hunk 1 above)
Comment (sidvishnoi) [reply routes/xref/lib/search.ts:149]: Didn't get it. Can you write an example/suggestion?

Comment (marcoscaceres) [reply routes/xref/lib/search.ts:149]: Something like: 

```JS
return termGenerator(term, {single: shouldTreatAsConcept}) : 

function* termGenerator(term){
   yield term;
   if(!single) yield* textVariations(term); 
}
```

Comment (sidvishnoi) [reply routes/xref/lib/search.ts:149]: Decided over Slack to keep as is.

**Pull request merged**